spec/ruby_terraform/commands/apply_spec.rb

```diff
@@ -1,7 +1,10 @@
 require 'spec_helper'
 
 describe RubyTerraform::Commands::Apply do
+  let(:command) { described_class.new(binary: 'terraform') }
+
   before(:each) do
+    allow(Open4).to receive(:spawn)
```

Comment:
Is there any way this could be handled instead by each shared behaviour?

Reply:
The reason being is it feels like this is "action at a distance" in that you affect something down inside the shared behaviours. Better to see the stubbing in context, even if that means stubbing the same thing multiple times.

Typically, I try to keep `before` and `after` as minimal as possible since any side effects introduced are effectively "global" to all tests and it makes it harder to understand the specific test context. Better to have the test itself define what it needs as a prerequisite within the test and pull the logic of achieving the prerequisite into a function.

Reply:
Good call - I've moved it into the lowest level shared example (`a valid command line`) that pretty much everything calls.  Made me realise I'd obviously got a little lost in the weeds refactoring the workspace spec and forgot I could use the `a valid command line` for a few of them so no need for the manual tests I had left in there now either..